chartfx-chart/src/main/java/de/gsi/chart/plugins/measurements/AbstractChartMeasurement.java

```diff
@@ -115,8 +115,12 @@ public void setDataSet(final DataSet dataSet) {
         }
 
         if (dataSet == null) {
-            valueField.setDataSetName("<unknown data set>");
-            this.dataSet = dataSet;
+            if (dataSetSelector.getNumberDataSets() == 1) {
+                valueField.setDataSetName(dataSetSelector.getFirstDataSet().getName());
+            } else {
+                valueField.setDataSetName("<unknown data set>");
+                this.dataSet = dataSet;
+            }
```

Comment:
This looks weird, from what I understand, this would set the name of the first data set, but leave the data set at its old value and it would also not setup the listener?
I suspect this only works for measurement plugins which do not really need the dataset anyway.
If the list of DataSets is not empty, dataSet should probably not be `null` here anyway imho.

(I noticed problems with DataSet selection in my App, too, but I haven't come around to look into it until now so thanks for bringing this up, i'll investigate)

Reply:
[These 3 lines](https://github.com/GSI-CS-CO/chart-fx/blob/master/chartfx-chart/src/main/java/de/gsi/chart/plugins/measurements/utils/DataSetSelector.java#L28-L30) seem to not have the intended effect.

Reply:
They seem to work since there is a solution in place in this method:

```
public DataSet getDataSet() {
        if (dataSet == null) {
            final List<DataSet> allDataSets = new ArrayList<>(chart.getAllDatasets());
            return allDataSets.get(0);
        }
        return dataSet;
}
```

Reply:
Still does not add the listener and it is not obvious to me that this will always return the same DataSet as the one whose name is shown in the box esp when datasets are added and removed frequently.

(I also have not yet looked into what the measurement implementations do with this dataset, esp since a DataSet might even be drawn by different renderers using different axes. But this is probably out of scope for this PR.)

Reply:
I do not understand the problem yet. I will think about it an reopen the PR when I have more insight.